src/main/java/bio/terra/common/db/DatabaseRetryUtils.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.common.db;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.RecoverableDataAccessException;
+import org.springframework.dao.TransientDataAccessException;
+
+/** Utilities to execute database operations with retry support. */
+public class DatabaseRetryUtils {
+  private static final Logger logger = LoggerFactory.getLogger(DatabaseRetryUtils.class);
+
+  /** Executes a database operation and retries if retryable. */
+  public static <T> T executeAndRetry(
+      DatabaseExecute<T> execute, Duration retrySleep, int maxNumRetries)
```

Comment:
naming nit: s/maxNumRetries/maxNumAttempts/?

If I'm reading correctly, with maxNumRetries=1, execution will only proceed once through before an InterruptedException is thrown, meaning zero retries were attempted. I think "attempts" as a parameter is less ambiguous. It's also more consistent with Spring Retry (https://www.baeldung.com/spring-retry)